test/infrastructure/container/docker.go

```diff
@@ -96,19 +96,19 @@ func (d *dockerRuntime) SaveContainerImage(ctx context.Context, image, dest stri
 // already exist. This is important when we're using locally build images in CI which
 // do not exist remotely.
 func (d *dockerRuntime) PullContainerImageIfNotExists(ctx context.Context, image string) error {
-	filters := dockerfilters.NewArgs()
-	filters.Add("reference", image)
-	images, err := d.dockerClient.ImageList(ctx, types.ImageListOptions{
-		Filters: filters,
-	})
+	imageExistsLocally, err := d.ImageExistsLocally(ctx, image)
 	if err != nil {
 		return fmt.Errorf("failure listing container images: %v", err)
 	}
-	// Nothing to do as the image already exists locally.
-	if len(images) > 0 {
+	if imageExistsLocally {
 		return nil
 	}
 
+	return d.PullContainerImage(ctx, image)
+}
+
+// PullContainerImage triggers the Docker engine to pull an image.
+func (d *dockerRuntime) PullContainerImage(ctx context.Context, image string) error {
```

Comment:
Is `PullContainerImage` supposed to get used externally or is it enough to have it private? (I don't know the use case in the provider tests :-), answer could be simply yes, there are use cases where it should be used instead of `PullContainerImageIfNotExists`).

Reply:
whoops, we need to modify `Runtime` (exported) interface in order to evolve the code in the simplest way possible, so yea, we need to be able to call it explicitly, we can't rely upon `PullContainerImageIfNotExists` in all cases

Reply:
Totally fine 👍  feel free to resolve :-)